### PR TITLE
Feature: Signup page / AccountDropdown

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/entities/user/lib/hooks/useFetchLogin.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  account: Account
+  authToken: Authorization
+  message: string
+  user: User
+}
+
+export type SuccessLoginResponse = AxiosResponse<ILoginSuccessResponse>
+export type FailedLoginResponse = AxiosError<BaseErrorResponse>
+
+export const useFetchLogin = (options: MutationOptions<SuccessLoginResponse, FailedLoginResponse, ILoginPayload>) => {
```
Comment (BamMironov) [inline src/entities/user/lib/hooks/useFetchLogin.ts:20]: I think you should come up with a different naming strategy for hooks that are intended for POST/PUT operations.
"Fetch" implies that we get/receive something. But it doesn't imply that we change or create something.
You can go with names, for example, "useLoginMutation" (my favorite) or "useLogin" or something similiar

File: src/entities/user/lib/hooks/useFetchLogin.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const name = "login"
+
+  const fetcher = useMutation([name], (data: ILoginPayload) => authorizationService.login(data), options)
```
Comment (BamMironov) [inline src/entities/user/lib/hooks/useFetchLogin.ts:23]: Here you can just return useMutation call right away

File: src/entities/user/lib/hooks/useFetchLogout.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import { authorizationService } from "~/entities/user/lib/authorization.service"
+import { BaseErrorResponse } from "~/utils/types/httpResponses"
+
+import { ILogoutPayload } from "../../model/api.interfaces"
+
+export type SuccessLogoutResponse = AxiosResponse<{ message: string }>
+export type FailedLogoutResponse = AxiosError<BaseErrorResponse>
+
+export const useFetchLogout = (
```
Comment (BamMironov) [inline src/entities/user/lib/hooks/useFetchLogout.ts:12]: Same here regarding the naming strategy

File: src/entities/user/lib/hooks/useFetchLogout.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  options: MutationOptions<SuccessLogoutResponse, FailedLogoutResponse, ILogoutPayload>,
+) => {
+  const name = "logout"
+
+  const fetcher = useMutation([name], (data: ILogoutPayload) => authorizationService.logout(data), options)
```
Comment (BamMironov) [inline src/entities/user/lib/hooks/useFetchLogout.ts:17]: Same here regarding the return

File: src/entities/user/lib/hooks/useFetchSignup.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+export interface ISignupSuccess extends User {
+  account: Account
+  authToken: Authorization
+}
+
+export type SuccessSignupResponse = AxiosResponse<ISignupSuccess>
+export type FailedSignupResponse = AxiosError<BaseErrorResponse>
+
+export const useFetchSignup = (
```
Comment (BamMironov) [inline src/entities/user/lib/hooks/useFetchSignup.ts:18]: Same here regarding the naming strategy

File: src/entities/user/lib/hooks/useFetchSignup.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  options: MutationOptions<SuccessSignupResponse, FailedSignupResponse, ISignupPayload>,
+) => {
+  const name = "signup"
+
+  const fetcher = useMutation([name], (data: ISignupPayload) => authorizationService.signup(data), options)
```
Comment (BamMironov) [inline src/entities/user/lib/hooks/useFetchSignup.ts:23]: Same here regarding the return

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/app/InactivityTracker.tsx [hunk 7]
```diff
@@ -17,20 +15,21 @@ const LOGOUT_TIME_LIMIT = 15 * ONE_MIN // 15 min
 
 export const InactivityTracker = ({ children }: InactivityTrackerProps) => {
   let timer: number | undefined
-  const dispatch = useAppDispatch()
   const navigate = useNavigate()
 
-  const { clearStorage } = useLocalStorage()
+  const { clearUserAndAuth, auth } = useAuth()
+  const mutation = useFetchLogout({})
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/InactivityTracker.tsx:21]: Done

File: src/app/InactivityTracker.tsx [hunk 8]
```diff
@@ -49,14 +48,15 @@ export const InactivityTracker = ({ children }: InactivityTrackerProps) => {
...
+  useEffect(() => {
     Object.values(events).forEach(item => {
       window.addEventListener(item, () => {
         resetTimer()
         logoutTimer()
       })
     })
-  })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/InactivityTracker.tsx:59]: Makes sense. Done

File: src/entities/user/lib/hooks/useFetchLogin.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/entities/user/lib/hooks/useFetchLogin.ts]: Done

File: src/entities/user/lib/hooks/useFetchLogin.ts (hunk 2 above)
Comment (moiskillnadne) [reply src/entities/user/lib/hooks/useFetchLogin.ts:25]: Done

File: src/entities/user/lib/hooks/useFetchLogout.ts (hunk 3 above)
Comment (moiskillnadne) [reply src/entities/user/lib/hooks/useFetchLogout.ts:12]: Done

File: src/entities/user/lib/hooks/useFetchLogout.ts (hunk 4 above)
Comment (moiskillnadne) [reply src/entities/user/lib/hooks/useFetchLogout.ts:17]: Done

File: src/entities/user/lib/hooks/useFetchSignup.ts (hunk 5 above)
Comment (moiskillnadne) [reply src/entities/user/lib/hooks/useFetchSignup.ts:18]: Done

File: src/entities/user/lib/hooks/useFetchSignup.ts (hunk 6 above)
Comment (moiskillnadne) [reply src/entities/user/lib/hooks/useFetchSignup.ts:23]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**